src/wallet/crypter.cpp

```diff
@@ -151,6 +151,23 @@ static bool DecryptSpendingKey(const CKeyingMaterial& vMasterKey,
     return sk.address() == address;
 }
 
+static bool DecryptSaplingSpendingKey(const CKeyingMaterial& vMasterKey,
+                               const std::vector<unsigned char>& vchCryptedSecret,
+                               const libzcash::SaplingFullViewingKey& fvk,
+                               libzcash::SaplingSpendingKey& sk)
+{
+    CKeyingMaterial vchSecret;
+    if(!DecryptSecret(vMasterKey, vchCryptedSecret, fvk.GetFingerprint(), vchSecret))
+        return false;
+
+    if (vchSecret.size() != libzcash::SerializedSaplingSpendingKeySize)
+        return false;
+
+    CSecureDataStream ss(vchSecret, SER_NETWORK, PROTOCOL_VERSION);
```

Comment:
Is this actually dependent on the network protocol version? That doesn't sound right to me.

Reply:
It matches what we do in e.g. `src/key_io.h`. The distinction from upstream, as I understand it, is that anything that might get communicated spatially between nodes is technically part of the protocol, whereas anything that is only communicated temporally between a single node's past and future selves is `SER_DISK, CLIENT_VERSION`.

Reply:
I think this is questionable because, if we were to substantially change the network protocol, that would have to be versioned and upgraded quite differently from changing the external key formats. In any case, it's ok for this PR.